Machine learning guide 10.13 (#296)
SoftwareAG/c8y-docs: content/machine-learning/casting-defect-detection-bundle/prerequisites.md

  - /predictive-analytics/casting-defect-detection/#prerequisites
---

Download the [CastingDefectDetectionDemoProject.zip](/files/zementis/CastingDefectDetectionDemoProject.zip) file which contains the following files:

> * **Method1 files**

Reviewer: ```suggestion
* **Method 1 files**
```

  * Training dataset (*casting_data.zip*)
  * Pre-processing and post-processing Python scripts (*castingPreProcessingForNN.py* and *castingPostProcessingForNN.py*)
  * Test images (*testDefectImage.png* and *testOkImage.png*)

> * **Method2 files**

Reviewer: ```suggestion
* **Method 2 files**
```

  * Training dataset (*casting_data.zip*)
  * Pre-processing and post-processing Python scripts (*castingPreProcessingForJNB.py* and *castingPostProcessingForJNB.py*)
  * Jupyter Notebook (*castingDefectDetectionDemo.ipynb*)
  * Test images (*testDefectImage.png* and *testOkImage.png*)

Running the demo scripts requires:

* Prior experience with Python and understanding of the data science processes.
* Familiarity with {{< product-c8y-iot >}} and its in-built apps.

> * Subscription of MLW, Zementis, ONNX microservices (10.13.0.x.x or higher) and Machine Learning Workbench, as well as the Machine Learning applications on the tenant.

Reviewer 2: ```suggestion
* Subscription of the MLW microservice, the Zementis microservice, the ONNX microservice (10.13.0.x.x or higher), the Machine Learning Workbench application and the Machine Learning application on the tenant.
```
